api-tester/src/main/java/com/revenuecat/apitester/java/DeprecatedPurchasesAPI.java

```diff
@@ -0,0 +1,114 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+@SuppressWarnings({"unused"})
+final class DeprecatedPurchasesAPI {
+    static void check(final Purchases purchases,
+                      final Activity activity,
+                      final SkuDetails skuDetails,
+                      final StoreProduct storeProduct,
+                      final Package packageToPurchase,
+                      final UpgradeInfo upgradeInfo) {
+        final ArrayList<String> skus = new ArrayList<>();
+
+        final ReceiveOfferingsListener receiveOfferingsListener = new ReceiveOfferingsListener() {
+            @Override public void onReceived(@NonNull Offerings offerings) {}
+            @Override public void onError(@NonNull PurchasesError error) {}
+        };
+        final GetSkusResponseListener skusResponseListener = new GetSkusResponseListener() {
+            @Override public void onReceived(@NonNull List<? extends SkuDetails> skus) {}
+            @Override public void onError(@NonNull PurchasesError error) {}
+        };
+        final ProductChangeListener productChangeListener = new ProductChangeListener() {
+            @Override public void onCompleted(@Nullable Purchase purchase, @NonNull PurchaserInfo customerInfo) {}
+            @Override public void onError(@NonNull PurchasesError error, boolean userCancelled) {}
+        };
+        final MakePurchaseListener makePurchaseListener = new MakePurchaseListener() {
+            @Override public void onCompleted(@NonNull Purchase purchase, @NonNull PurchaserInfo customerInfo) {}
+            @Override public void onError(@NonNull PurchasesError error, boolean userCancelled) {}
+        };
+        final ReceivePurchaserInfoListener receiveCustomerInfoListener = new ReceivePurchaserInfoListener() {
+            @Override public void onReceived(@NonNull PurchaserInfo customerInfo) {}
+            @Override public void onError(@NonNull PurchasesError error) {}
+        };
+
+        purchases.getOfferings(receiveOfferingsListener);
+        purchases.getSubscriptionSkus(skus, skusResponseListener);
+        purchases.getNonSubscriptionSkus(skus, skusResponseListener);
+
+        final ProductChangeCallback productChangeCallback = new ProductChangeCallback() {
+            @Override public void onCompleted(@Nullable StoreTransaction storeTransaction, @NonNull CustomerInfo customerInfo) { }
+            @Override public void onError(@NonNull PurchasesError error, boolean userCancelled) {}
+        };
+
+        final PurchaseCallback purchaseCallback = new PurchaseCallback() {
+            @Override public void onCompleted(@Nullable StoreTransaction storeTransaction, @NonNull CustomerInfo customerInfo) { }
+            @Override public void onError(@NonNull PurchasesError error, boolean userCancelled) {}
+        };
+
+        purchases.purchaseProduct(activity, skuDetails, upgradeInfo, productChangeListener);
```

Comment:
we can use lambdas in java now that we're on 8 right? think there's any benefit to having the API tester include lambda versions too? (same question for kotlin)

Reply:
`purchaseProduct` doesn't have a version with lambdas right? I know we could do that if `ProductChangeListener` was a functional interface, but it's not because it has more than one method